docs/module_web_with_python.md

```diff
@@ -5,137 +5,437 @@
 #### What is DOM and how to manipulate it from Javascript?
+* DOM stands for Document Oject Model for html and XML.
+* DOM represents the document structure, and style, content.
+* Javascript uses DOM objects to manipulate a page, to add/remove/or modify elements to/of a page.
+* Individual elements can be made variable of in Javascript by using "document.getElementById('id')" or similar built in functions.
+* New DOM elements can be created via "document.createElement('')" function, which then can be added by using "element.appendChild(newStuff)".
 #### What are events and how/why to use them in Javascript?
+* Events are acting as triggers for certain functions in a javascript code.
+* Events can also be used as targets to target DOM elements in a Javascript function via "event.target"
+* examples for events: "click", "dragover", "dragstart", "dragend"
 #### What is event bubbling/capturing? How would you use it?
+* Event bubbling happens when an inner scope function is called while an outer scope DOM element also has an event listener tied to it on the same event trigger.
+* Event bubbling means that all functions will activate one by one starting from the innermost scope.
+* Capturing is done to prevent the events from triggering all at once.
+* example: event.cancelBubble = true; or event.stopPropagation();
 #### What is JSON and how do we use it?
-
+* JSON stands for JavaScript Object Notation.
+* JSON is an open standard  file format and data interchange format.
+* JSON files and data consist of key-value pairs but are not to be mistaken with DICT objects.
+* JSON data can be used to communicate with the server using JavaScript.
+* JSON data can be used to deliver data to and from the server without the need of reloading a webpage.
+* JSON files can be used to store settings in applications.
 ## Software engineering
 
 ### Version control
 
 #### What type of branching strategy would you use?
+* I would make a different branch for every task that are not closely related and name the branches according to the company guidelines.
 #### What would you do if you find a bug on the production code (master branch)?
+* Roll back the merge and continue to work on the feature branch.
+* Start a new branch for fixing the bug.
 #### How can you move changes from one branch to another in GIT?
+* By creating pull requests and merging the two branches that I want.
 #### How does a VCS help with code reviews?
+* VCS provides standard requirements that all projects must follow.
 #### What is your favorite git command? Why?
+* Commit, because when I'm using it, it means that I've progressed with my tasks.
 #### What does remote/local mean in Git? 
+* Local means files and changes that are only on your computer and are not shared.
+* Remote are the files and changes that are shared between every collaborator in the project.
 
 ### DevOps
 
 #### Why is it good to use a package manager software?
+* Installing softwares are made way easier by their existence.
```

Comment:
Typo "softwares are" -> "software is". Optional.

Reply:
fixed